app/src/editor/canvas/components/CodeEditor/index.jsx

```diff
@@ -1,101 +1,114 @@
-import React from 'react'
+import React, { useState, useCallback } from 'react'
 
 import CodeEditorWindow from './CodeEditorWindow'
 import DebugWindow from './DebugWindow'
 
-export default class CodeEditor extends React.Component {
-    state = {
-        editorOpen: false,
-        debugOpen: false,
-        editorPosition: {
-            x: 0,
-            y: 0,
-        },
-        debugPosition: {
-            x: 0,
-            y: 0,
-        },
-    }
+export const CodeEditor = ({
+    children,
+    code,
+    readOnly,
+    onApply,
+    onChange,
+    debugMessages,
+    onClearDebug,
+}) => {
+    const [editorOpen, setEditorOpen] = useState(false)
+    const [debugOpen, setDebugOpen] = useState(false)
+    const [editorLayout, setEditorLayout] = useState({
+        x: 0,
+        y: 0,
+        width: 600,
+        height: 400,
+    })
+    const [debugLayout, setDebugLayout] = useState({
+        x: 0,
+        y: 0,
+        width: 600,
+        height: 400,
+    })
+    const onShowEditor = useCallback(() => {
+        setEditorOpen(true)
+    }, [setEditorOpen])
```

Comment:
`setEditorOpen` as well as any other setter coming from `useState` doesn't have to be in the deps array. It always is the same function.

Reply:
Same applies to other `useCallback` here.